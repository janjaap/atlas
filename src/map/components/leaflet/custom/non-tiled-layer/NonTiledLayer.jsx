import { TileLayer } from 'react-leaflet';
import L from 'leaflet';
import queryString from 'querystring';
import 'leaflet.nontiledlayer';

class NonTiledLayer extends TileLayer {
  createLeafletElement(props) {
    const { url, params, ...args } = props;

    const query = (params) ? queryString.stringify(params) : '';
    const layerUrl = (query) ? `${url}?${query}` : url;

    return L.nonTiledLayer.wms(layerUrl, this.getOptions(args));
  }

  updateLeafletElement(fromProps, toProps) {
    super.updateLeafletElement(fromProps, toProps);
    if (toProps.params !== fromProps.params) {
      this.leafletElement.setParams(toProps.params);
    }
  }
}

export default NonTiledLayer;
