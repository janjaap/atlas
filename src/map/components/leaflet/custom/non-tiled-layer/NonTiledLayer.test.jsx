import React from 'react';
import { shallow } from 'enzyme';
import NonTiledLayer from './NonTiledLayer';

function renderComponent() {
  return shallow(
    <NonTiledLayer
      url="url"
      params={{ test: '123' }}
    />, {
      context: {
        layerContainer: {
          addLayer: jest.fn()
        }
      }
    }
  );
}

describe('NonTiledLayer', () => {
  const props = {
    opacity: 100,
    url: 'https://pano.amsterdam.nl',
    params: {
      mission_type: 'bi',
      mission_year: 2017
    }
  };

  it('should render everything', () => {
    const wrapper = renderComponent();
    wrapper.instance().createLeafletElement({ props });

    expect(wrapper).toMatchSnapshot();
  });

  it('should rerender everything when props change', () => {
    const wrapper = renderComponent();
    const setParamsMock = jest.fn();

    wrapper.instance().leafletElement = {
      setParams: setParamsMock,
      setOpacity: jest.fn(),
      setUrl: jest.fn()
    };

    wrapper.instance().updateLeafletElement({ props }, { props });
    expect(setParamsMock).not.toHaveBeenCalled();

    wrapper.instance().updateLeafletElement({ props }, { ...props, params: { mission_type: 'WOZ', mission_year: 2018 } });
    expect(setParamsMock).toHaveBeenCalled();
  });
});
