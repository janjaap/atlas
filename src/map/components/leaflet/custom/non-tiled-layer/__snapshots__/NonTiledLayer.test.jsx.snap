// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NonTiledLayer should render everything 1`] = `""`;
